### PR TITLE
[qa] Get rid of duplicate code

### PR DESCRIPTION
**Pull request by maflcko**
This removes some unused and duplicate code in maxuploadtarget.py

### CODE REVIEW

**Comment by sdaftuar** [comment]
Perhaps you could also do a similar replacement for pruning.py, which uses this same duplicated mine_full_block() function?

File: qa/rpc-tests/maxuploadtarget.py [hunk 1]
```diff
@@ -162,11 +129,11 @@ def run_test(self):
         self.nodes[0].setmocktime(int(time.time()) - 2*60*60*24)
 
         # Mine one more block, so that the prior block looks old
-        self.mine_full_block(self.nodes[0], self.nodes[0].getnewaddress())
+        mine_large_block(self.nodes[0])
 
         # We'll be requesting this new block too
         big_new_block = self.nodes[0].getbestblockhash()
-        new_block_size = self.nodes[0].getblock(big_new_block)['size']
+        # print(self.nodes[0].getblock(big_new_block)['size'])
```
Comment (mrbandrews) [inline qa/rpc-tests/maxuploadtarget.py]: nit: think you left this in accidentally

File: qa/rpc-tests/pruning.py [hunk 2]
```diff
@@ -95,17 +85,17 @@ def create_chain_with_staleblocks(self):
             stop_node(self.nodes[0],0)
             self.nodes[0]=start_node(0, self.options.tmpdir, ["-debug","-maxreceivebuffer=20000","-blockmaxsize=999000", "-checkblocks=5"], timewait=900)
             # Mine 24 blocks in node 1
-            self.utxo = self.nodes[1].listunspent()
+            utxo = self.nodes[1].listunspent()
```
Comment (mrbandrews) [inline qa/rpc-tests/pruning.py]: nit: can remove this line

File: qa/rpc-tests/pruning.py [hunk 3]
```diff
@@ -95,17 +85,17 @@ def create_chain_with_staleblocks(self):
...
             for i in range(24):
                 if j == 0:
-                    self.mine_full_block(self.nodes[1],self.address[1])
+                    mine_large_block(self.nodes[1])
                 else:
                     self.nodes[1].generate(1) #tx's already in mempool from previous disconnects
 
             # Reorg back with 25 block chain from node 0
-            self.utxo = self.nodes[0].listunspent()
+            utxo = self.nodes[0].listunspent()
```
Comment (mrbandrews) [inline qa/rpc-tests/pruning.py]: same

**Comment by mrbandrews** [comment]
A couple of nits, otherwise ACK

**Comment by MarcoFalke** [comment]
Squashed and fixed nits: fa130b8..facbfa5

**Comment by morcos** [comment]
@MarcoFalke @sdaftuar @mrbandrews Did you test this?
It seems to have slowed pruning.py by 35%.  I don't think its worth it at the expense of adding 10 mins to the pruning.py test.

**Comment by MarcoFalke** [comment]
Jup, I was running the tests after squash. Will take a look at the
slowdown. Should be easy to fix if this commit caused it.

**Comment by morcos** [comment]
@marcofalke my guess is it's the repeated calling of listunspent instead of caching the utxo's.  well I should say that's the first difference I saw and then was too tired to investigate further.

**Pull request merged**